overlord/devicestate/handlers.go

```diff
@@ -790,3 +795,134 @@ func fetchKeys(st *state.State, keyID string) (errAcctKey error, err error) {
 		keyID = a.SignKeyID()
 	}
 }
+
+func snapState(st *state.State, name string) (*snapstate.SnapState, error) {
+	var snapst snapstate.SnapState
+	err := snapstate.Get(st, name, &snapst)
+	if err != nil && err != state.ErrNoState {
+		return nil, err
+	}
+	return &snapst, nil
+}
+
+func makeRollbackDir(name string) (string, error) {
+	rollbackDir := filepath.Join(dirs.SnapRollbackDir, name)
+
+	if err := os.MkdirAll(rollbackDir, 0755); err != nil {
```

Comment:
How much do we care about protecting this dir? Should this be `0750` ?

Reply:
Good question. The update runner actually checks that the data in the snap is different from what's on the disk. In most cases, the content will probably be originated from the snap anyway, so it's already accessible globally. I don't mind going with 0750 though.